frontend/pages/queries/QueryPage/components/QueryForm/QueryForm.tsx

```diff
@@ -48,6 +49,28 @@ const validateQuerySQL = (query: string) => {
   return { valid, errors };
 };
 
+// https://usehooks.com/useDebounce/
+const useDebounce = (value: number | string, delay: number) => {
+  // State and setters for debounced value
+  const [debouncedValue, setDebouncedValue] = useState(value);
+  useEffect(
+    () => {
+      // Update debounced value after delay
+      const handler = setTimeout(() => {
+        setDebouncedValue(value);
+      }, delay);
+      // Cancel the timeout if value changes (also on delay change or unmount)
+      // This is how we prevent debounced value from updating if value is changed
+      // within the delay period. Timeout gets cleared and restarted.
+      return () => {
+        clearTimeout(handler);
+      };
+    },
+    [value, delay] // Only re-call effect if value or delay changes
+  );
+  return debouncedValue;
+};
+
```

Comment:
Would this function be helpful to other components? If so, consider moving to `fleet/helpers.ts`.

Reply:
@lukeheath yes I think this could be a helpful one generally. Perhaps we might want to have a separate set of utilities just for custom hooks like this one? That `fleet/helpers` file is getting rather unwieldy in my opinion.

Reply:
Agreed it is getting pretty big. A new custom hooks file sounds great.

Reply:
@lukeheath I've refactored with the `use-debounce` library. Please give it a try and let me know if you think it is performing well. Thanks!